core/src/main/java/org/mule/routing/AbstractAggregator.java

```diff
@@ -148,27 +148,20 @@ protected Factory internalEventsGroupsObjectStoreFactory()
             @Override
             public Object create()
             {
-                try
+                ObjectStore objectStore;
+                if (persistentStores)
+                {
+                    objectStore = muleContext.getRegistry().lookupObject(OBJECT_STORE_DEFAULT_PERSISTENT_NAME);
```

Comment:
This must be done using the ObjectStoreManager, just ask for a persistent object store.